### PR TITLE
feat: refactor and generalize propagation validation

### PR DESCRIPTION
**Pull request by asalzburger**
This supersedes #3434 and rebases it onto the latest updates of the propagation example.

This changes the output from a simple step collection from the propagation example to a `PropagationSummary` object,
which contains the input parameters and the step collections. 

- with this, the `RootPropgationStepWriter` and the new `RootPropagationSummaryWriter` can now easily write out information of the propagation job and relate them to the start parameters

- writing out the `PropagationSummary` will eventually allow to have a `DirectPropagationAlgorithm` (version)  and a `TryAllPropagationAlgorithm` (version) which can sandwich the `PropagationAlgorithm` and will give us information about:

- potential navigation failures versus brute-force navigation
- overhead of navigation wrt direct propagation

### CODE REVIEW

**Comment by paulgessinger** [comment]
@asalzburger can you describe what this is about? 

- What does this do?
- Why is it needed?

**Comment by paulgessinger** [comment]
Also, I think @andiwand wanted to circle back around to this. He should be back Monday, surely we can wait until then with this?

**Comment by asalzburger** [comment]
> Also, I think @andiwand wanted to circle back around to this. He should be back Monday, surely we can wait until then with this?

Yes, no problem, I though I would take some work off his shoulders, because the main rebasing is coming from the input change to particle gun, which I authored.

File: Examples/Io/Root/src/RootPropagationStepsWriter.cpp [hunk 1]
```diff
@@ -55,15 +57,27 @@ ActsExamples::RootPropagationStepsWriter::RootPropagationStepsWriter(
...
+  m_outputTree->Branch("t", &m_t);
+  m_outputTree->Branch("eta", &m_eta);
+  m_outputTree->Branch("pt", &m_pt);
+  m_outputTree->Branch("p", &m_p);
   m_outputTree->Branch("volume_id", &m_volumeID);
   m_outputTree->Branch("boundary_id", &m_boundaryID);
   m_outputTree->Branch("layer_id", &m_layerID);
   m_outputTree->Branch("approach_id", &m_approachID);
   m_outputTree->Branch("sensitive_id", &m_sensitiveID);
+  m_outputTree->Branch("sensitive_id", &m_extraID);
```
Comment (andiwand) [inline Examples/Io/Root/src/RootPropagationStepsWriter.cpp:75]:
```suggestion
  m_outputTree->Branch("extra_id", &m_extraID);
```

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [7c272316b739a72bf45d92e5eb336295e5b41201]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42586955) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7973220)!

**Pull request merged**